From a48e1bc6b70a79d672a994ff05f28f48bcb49e14 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 1 Jun 2017 13:13:52 +0500
Subject: [PATCH] CMakeList: only search for jsoncpp when needed

JSONCPP is only required for the case when VIA
is enabled so make the library/include searches
conditional.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6fc30b8c0..d81a9e71a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -217,6 +217,7 @@ find_path(SPIRV_TOOLS_INCLUDE_DIR spirv-tools/libspirv.h HINTS "${EXTERNAL_SOURC
 
 find_path(X11_XLIB_INCLUDE_DIR X11/Xlib.h DOC "Path to X11/Xlib.h")
 
+if(BUILD_VIA)
 find_path(JSONCPP_INCLUDE_DIR json/json.h HINTS "${EXTERNAL_SOURCE_ROOT}/jsoncpp/dist"
                                                    "${EXTERNAL_SOURCE_ROOT}/JsonCpp/dist"
                                                    "${EXTERNAL_SOURCE_ROOT}/JsonCPP/dist"
@@ -230,6 +231,7 @@ find_path(JSONCPP_SOURCE_DIR jsoncpp.cpp HINTS "${EXTERNAL_SOURCE_ROOT}/jsoncpp/
                                                    "${EXTERNAL_SOURCE_ROOT}/JSONCPP/dist"
                                                    "${CMAKE_SOURCE_DIR}/../jsoncpp/dist"
                                              DOC "Path to jsoncpp/dist/json.cpp")
+endif()
 
     find_library(GLSLANG_LIB NAMES glslang
         HINTS ${GLSLANG_SEARCH_PATH} )
@@ -252,8 +254,10 @@ find_path(JSONCPP_SOURCE_DIR jsoncpp.cpp HINTS "${EXTERNAL_SOURCE_ROOT}/jsoncpp/
     find_library(SPIRV_TOOLS_LIB NAMES SPIRV-Tools
              HINTS ${SPIRV_TOOLS_SEARCH_PATH} )
 
+if(BUILD_VIA)
     find_library(JSONCPP_LIB NAMES jsoncpp
              HINTS ${JSONCPP_SEARCH_PATH} )
+endif()
 
 if (WIN32)
     add_library(glslang     STATIC IMPORTED)
-- 
2.11.1

